code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,206 @@
+		src.is_npc = TRUE
+		START_TRACKING
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new/datum/limb/sawfly_blades
+		HH.name = "sawfly blades"
+		HH.limb_name = HH.name
+		HH.can_hold_items = FALSE
+		HH.can_range_attack = TRUE
+
+	disposing()
+		. = ..()
+		STOP_TRACKING
+
+
+	proc/foldself()
+		if(!isalive(src))
+			return 0
+		else
+			var/obj/item/old_grenade/sawfly/reused/N = new /obj/item/old_grenade/sawfly/reused(get_turf(src))
+			// pass our name and health
+			N.name = "Compact [name]"
+			N.tempname = src.name
+			N.tempdam = (50 - src.health )
+			qdel(src)
+
+
+	proc/communalbeep() // distributes the beepchance among the number of sawflies nearby
+		fliesnearby = 1 //that's you, little man! :)
+		for_by_tcl(E, /mob/living/critter/robotic/sawfly)
+			if(isalive(E))
+				src.fliesnearby += 1 //that's your buddies!
+		var/beepchance = (1 / fliesnearby) * 100 //if two sawflies, give 50% chance that any one will beep
+		if(fliesnearby<3) beepchance -=20 //heavily reduce chance of beep in swarm
+		if(prob(beepchance))
+			if(isalive(src))
+				playsound(src, pick(src.beeps), 40, 1)
+				src.visible_message("<b>[src] [beeptext].</b>")
+
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to spiral out of control!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	Cross(atom/movable/mover) //code that ensures projectiles hit them when they're alive, but won't when they're dead
+		if(istype(mover, /obj/projectile))
+			if(!isalive(src))
+				return 1
+		return ..()
+
+//note: due to the AIholder's timed nature, they can still priority attack you if you're already targeted, but it's incredibly rare. Frankly I think it adds to the challenge.
+//doublenote: the absolute agony that was trying to get this to function in any way that wasn't incredibly obtuse and hacky without going back to the projectile.
+	attackby(obj/item/W as obj, mob/living/user as mob)
+		if(!(istraitor(user) || isnukeop(user) || isspythief(user) || (user in src.friends)) || (user.health < 40))//are you an eligible target: nonantag or healthy enough?
+			if(prob(50) && isalive(src))//now that you're eligible, are WE eligible?
+				if((ai.target != user))
+					ai.interrupt()//even though the AI doing this is nigh impossible, we'll still want to tell the AI that something's happening
+					src.visible_message("<span class='alert'><b>[src]'s targeting subsystems identify [user] as a high priority threat!</b></span>")
+					//first attack is with the hand, so the sawfly can't triple attack if it was just now harming somone
+					src.set_dir(get_dir(src, user))
+					src.hand_attack(user, dummy_params)
+					//second attack is hardcoded, since the limb has a cooldown of 1 seconds between attacks that interferes otherwise
+					if(isalive(src) && (get_dist(src, user) > 1)) //account for SPAWN() jank
```

Comment:
wrong way round, these checks should be *inside* the `SPAWN()` since you want them to run *after* the 5 second pause.